### DISCUSSION TITLE
Portenta _H7_AsyncWebServer crashing, stack overflow?

### DISCUSSION THREAD

**Discussion by bsundahl1** [khoih-prog/Portenta_H7_AsyncWebServer#14] [Q&A]
All of the Portenta_H7_AsyncWebServer examples are crashing on me. Most of them start out working, but after a few minutes, the Portenta will crash with a blinking LED code (4 long / 4 short). It appears that the crash occurs during the write to the browser (Chrome). Any of these examples will run all day if I do not make the connection from the browser. It always seems to crash on a request->send() call, but not always the same one. Sometimes it is in handleRoot() and sometimes in drawGraph().

The worst case is "Async_AdvancedWebServer_MemoryIssues_Send_CString.ino", which will crash every time on the first call to drawGraph request->send(). It does not seem to matter how much data I write.

I did figure out how to check the stack sizes, and it does appear that one process may be overflowing its allocated stack space. But I might be misinterpreting this. This is maximum stack size as measured by mbed, vers the reserved stack space.

// asynchronous WebServer  https://github.com/khoih-prog/Portenta_H7_AsyncWebServer

    Thread ID: 2400AFF4 Stack size (max/res): 1208 / 32768
    Thread ID: 2400B900 Stack size (max/res): 1200 / 1200
    Thread ID: 240028A0 Stack size (max/res): 328 / 896
    Thread ID: 2400DA40 Stack size (max/res): 384 / 2048
    Thread ID: 2400285C Stack size (max/res): 104 / 768
    Thread ID: 24011EB8 Stack size (max/res): 128 / 256
    Thread ID: 2400B3E4 Stack size (max/res): 200 / 2048

If I use the "non-async" EthernetWebServer library, it will run all day without issue.
I see the same processes running, but the max stack sizes are quite different.

// normal WebServer   https://github.com/khoih-prog/EthernetWebServer

    Thread ID: 2400BF8C Stack size (max/res): 1848 / 32768
    Thread ID: 2400C284 Stack size (max/res): 704 / 1200
    Thread ID: 24003838 Stack size (max/res): 328 / 896
    Thread ID: 2400EBB0 Stack size (max/res): 472 / 2048
    Thread ID: 240037F4 Stack size (max/res): 104 / 768
    Thread ID: 24011808 Stack size (max/res): 128 / 256
    Thread ID: 2400C16C Stack size (max/res): 200 / 2048

Is there something I could be doing to cause this?
I am using a Portenta H7 Lite with Portenta Breakout.
I am using a cabled connection to my router.
I am using PlatformIO and the crazy long platformio.ini file that is included with the library, and building with the default environment (which builds for the M7 core).
I am using Portenta_H7_AsyncTCP v1.4.0  and Portenta_H7_AsyncWebServer v1.4.2
I had seen this same issue using Portenta_H7_AsyncWebServer v1.2.1
I am using a cable connection to my router.

Am I the only one that has this problem?

**Comment by khoih-prog** [comment]
> I am using PlatformIO

Try with Arduino IDE to see if this is issue with PIO ???

> I am using a Portenta H7 Lite with Portenta Breakout.

If OK, then try with another board, or with normal Portenta_H7 (not Lite) and/or without Portenta Breakout.

I'm afraid no one can help you if the issue persists (environment, settings, hardware, etc.)

Ask on Arduino Forum or even Arduino core to see anybody can help with your strange problem. 

Good Luck,
